src/main.rs

```diff
@@ -273,32 +281,27 @@ impl<'a> Frame<'a> {
 
     }
 
+    pub fn register_class(&mut self, name: String, methods: HashMap<String,Node>) {
+        self.vm.classes.insert(name, Class::new(methods));
+    }
+
+
 }
 
+mod parse;
+
 
 fn main() {
-    let exp = Node::Addition(Box::new(Node::Integer(7)), Box::new(Node::Integer(2)));
-    let if_exp = Node::If{cond:Box::new(Node::Bool(false)), body: Box::new(Node::Integer(3))};
-
-    // Test block with assignment
-    let mut stmts = Vec::new();
-
-    let new_obj = Node::NewObj{class_name: String::from("Obj"), fields: None};
-    let setter = Node::SetObjField{
-        var: Box::new(Node::Var(String::from("x"))),
-        field: String::from("hello"),
-        value: Box::new(Node::Integer(10))
-    };
-    let getter = Node::GetObjField{
-        var: Box::new(Node::Var(String::from("x"))),
-        field: String::from("hello"),
-    };
-    stmts.push(Node::Let(String::from("x"), Box::new(new_obj)));
-    stmts.push(setter);
-    stmts.push(getter);
-    let block = Node::Block(stmts);
-    let mut vm = VM::new();
-    let res = vm.interp(&block);
-    println!("{}", res)
+
+
+    //FIXME: Unhardcode these paths
```

Comment:
These need to be fixed before merging.

Reply:
Fixed and allowed source prog to be passed as CLI